app/scripts/lib/metrics.js

```diff
@@ -152,7 +152,7 @@ define([
       var url = this._collector + '/metrics';
       var self = this;
 
-      return this._send(filteredData, url)
+      return this._send(filteredData, url, ! flushSynchronously)
```

Comment:
> am I missing something?

I doubt you are missing something. This sounds like a bigger task than I expected, do you think it is worthwhile to open a new PR and merge this one as is?